libraries/AP_RCProtocol/AP_RCProtocol.cpp

```diff
@@ -589,6 +600,23 @@ bool AP_RCProtocol::protocol_enabled(rcprotocol_t protocol) const
     return ((1U<<(uint8_t(protocol)+1)) & rc_protocols_mask) != 0;
 }
 
+#if AP_RCPROTOCOL_MAVLINK_RADIO_ENABLED
+void AP_RCProtocol::handle_radio_rc_channels(const mavlink_radio_rc_channels_t* packet)
+{
+    // take a shortcut if protocol is known to be MAVLINK_RADIO
+    if (_detected_protocol == AP_RCProtocol::MAVLINK_RADIO) {
+        backend[_detected_protocol]->update_radio_rc_channels(packet);
+        return;
+    }
+
+    for (uint8_t i = 0; i < ARRAY_SIZE(backend); i++) {
```

Comment:
Why pass to the other backends if we know only the new backend can handle the message?

Reply:
That is to say why not:
```
if (backend[AP_RCProtocol::MAVLINK_RADIO] == nullprt) {
    return;
}
backend[AP_RCProtocol::MAVLINK_RADIO]->update_radio_rc_channels(packet);
```
I'm not familiar with the RC protocol handling, maybe I'm missing something.

Reply:
hm, good question. I think I initially thought about something along this lines but somehow concluded this won't work, but thinking about your comment I now have a feeling that I may have gotten it wrong. Let me pl rethink and double check.

Reply:
could also avoid the special case for _detcted_protocol at the start of the fn too

Reply:
@IamPete1 
many thx again for the comment. It works beautifully, I feel a bit embarrassed to not have seen that option. With all the masks and #ifs I was worried a situation could exits where the fixed index would go beyond range, but this can't happen. Much nicer and cleaner now. Thx. 👍